synthesis/codon/analyzer.go

```diff
@@ -0,0 +1,36 @@
+package codon
+
+import "github.com/TimothyStiles/poly/io/genbank"
+
+// Stats denotes a set of computed codon table statistics
+type Stats struct {
+	StartCodonCount map[string]int `json:"start_codons_counts"`
+}
+
+func NewStats() *Stats {
+	return &Stats{
+		StartCodonCount: map[string]int{},
+	}
+}
+
+// Analyzer is used to compute codon usage statistics
+type Analyzer struct{}
+
+// ComputeStats returns a set of codon related statistics from a genebank sequence
+func (a *Analyzer) ComputeStats(sequence genbank.Genbank) (*Stats, error) {
+	stats := NewStats()
+
+	// iterate through the features of the genbank file and if the feature is a coding region, get the start codon
+	for _, feature := range sequence.Features {
+		if feature.Type == "CDS" {
+			sequence, err := feature.GetSequence()
```

Comment:
Wait, why does this method of `Feature` have an error in its return type? Reading through the implementation, the error will always be `nil`. @Koeng101 please confirm whether this is intended behavior, if not I will open a new issue for it.

Reply:
Not the intended behavior, though ironically I think most of the time its just panicked for me and I had to go fix the panic. The reason is that genbank location strings can be stupid, and impossible to parse.